### ISSUE TITLE
Support displaying grid lines

### ISSUE THREAD

**Issue by DougHennig** [ggreen86/XLSX-Workbook-Class#20]
You can turn off or on gridlines by adding showGridLines="0" or showGridLines="1" attributes in the &lt;sheetViews&gt; element of a sheet XML file. I suggest adding a property such as lDisplayGridLines to support that and handle it in all mechanisms that output to an XLSX file, including the SaveGrid* and SaveTable* methods.

**Comment by DougHennig** [comment]
I added a property named lDisplayGridLines and updated the following code in SaveGridToWorkbookEx:

```
*** 2018-08-22 DH: added support for This.lDisplayGridLines
***		IF tlFreeze
***			FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0">')
		IF tlFreeze or This.lDisplayGridLines
			FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0" ' + ;
				'showGridLines="' + iif(This.lDisplayGridLines, '1', '0') + '">')
			if tlFreeze
*** 2018-08-22 DH: end of new code
			FWRITE(lhFile, '<pane state="frozenSplit" activePane="bottomLeft" topLeftCell="A2" ySplit="1"/>')
			FWRITE(lhFile, '<selection activeCell="A2" sqref="A2" pane="bottomLeft"/>')
*** 2018-08-22 DH: added ENDIF to balance added IF
			endif
*** 2018-08-22 DH: end of new code
			FWRITE(lhFile, '</sheetView></sheetViews>')
		ENDIF
```

Similarly, I changed code in WriteSheetXMLs:

```
		DO CASE
			CASE xl_sheets.ysplit > 0 .AND. xl_sheets.xsplit = 0      && Freeze top row(s) only
*** 2018-08-22 DH: added support for This.lDisplayGridLines
***				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0">')
				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0" ' + ;
					'showGridLines="' + iif(This.lDisplayGridLines, '1', '0') + '">')
				lcRow = TRANSFORM(xl_sheets.ysplit+1)
				FWRITE(lhFile, '<pane state="frozenSplit" activePane="bottomLeft" topLeftCell="A' + lcRow + '" ySplit="' + TRANSFORM(xl_sheets.ysplit) + '"/>')
				FWRITE(lhFile, '<selection activeCell="A' + lcRow + '" sqref="A' + lcRow + '" pane="bottomLeft"/>')

			CASE xl_sheets.ysplit = 0 .AND. xl_sheets.xsplit > 0      && Freeze side column(s) only
*** 2018-08-22 DH: added support for This.lDisplayGridLines
***				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0">')
				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0" ' + ;
					'showGridLines="' + iif(This.lDisplayGridLines, '1', '0') + '">')
				lcColumn = this.ColumnIndexToAscii(xl_sheets.xsplit+1)
				FWRITE(lhFile, '<pane state="frozenSplit" activePane="bottomLeft" topLeftCell="' + lcColumn + '1" xSplit="' + TRANSFORM(xl_sheets.xsplit) + '"/>')
				FWRITE(lhFile, '<selection activeCell="' + lcColumn + '1" sqref="' + lcColumn + '1" pane="topRight"/>')

			CASE xl_sheets.ysplit > 0 .AND. xl_sheets.xsplit > 0      && Freeze both top row(s) and side column(s)
*** 2018-08-22 DH: added support for This.lDisplayGridLines
***				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0">')
				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0" ' + ;
					'showGridLines="' + iif(This.lDisplayGridLines, '1', '0') + '">')
				lcRow = TRANSFORM(xl_sheets.ysplit+1)
				lcColumn = this.ColumnIndexToAscii(xl_sheets.xsplit+1)
				FWRITE(lhFile, '<pane state="frozenSplit" activePane="bottomRight" topLeftCell="' + lcColumn + lcRow + ;
					'" ySplit="' + TRANSFORM(xl_sheets.ysplit) + '" xSplit="' + TRANSFORM(xl_sheets.xsplit) + '"/>')
				FWRITE(lhFile, '<selection activeCell="' + this.ColumnIndexToAscii(xl_sheets.xsplit) + '1" sqref="' + this.ColumnIndexToAscii(xl_sheets.xsplit) + '1" pane="topRight"/>')
				FWRITE(lhFile, '<selection activeCell="' + this.ColumnIndexToAscii(xl_sheets.xsplit) + lcRow + '" sqref="' + this.ColumnIndexToAscii(xl_sheets.xsplit) + lcRow + '" pane="bottomLeft"/>')
				FWRITE(lhFile, '<selection activeCell="' + lcColumn + lcRow + '" sqref="' + lcColumn + lcRow + '" pane="bottomRight"/>')

			OTHERWISE
*** 2018-08-22 DH: added support for This.lDisplayGridLines
***				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0"><selection activeCell="A1" sqref="A1"/>')
				FWRITE(lhFile, '<sheetViews><sheetView workbookViewId="0" ' + ;
					'showGridLines="' + iif(This.lDisplayGridLines, '1', '0') + ;
					'"><selection activeCell="A1" sqref="A1"/>')
*				FWRITE(lhFile, '<sheetViews><sheetView tabSelected="1" workbookViewId="0">')
		ENDCASE
```

**Comment by ggreen86** [comment]
I have added the capability to set the setting for showing/hiding grid lines.  However, I implemented it in a different manner.  Your solution would be global for all sheets in all workbooks that the class would create.  This might not be desired for all sheets.  So I have implemented in a manner to allow each sheet to be set independently for each workbook.  I also added to the open workbook methods to as to retain the setting in existing workbooks.  For grids the setting is provided as a parameter in the method or defaulted from the grid object setting.  This feature is included in Release 32.
